### DISCUSSION TITLE
Will you be adding all the genres Roku now lists?

### DISCUSSION THREAD

**Discussion by jsonnentag** [rrirower/Channel-Feed-Manager#169] [Q&A]
I was just going through things and checking every single entry (since no language has been required and the feed check gives an ERROR rather than an WARNING - or something non-critical) about it.  Oh, just for into, the Search Feed specification says it is OK to just have an entry at the top that says everything by default is 1 language and you don't have to have language entries for every single item.

Anyway, they now have a pile more of "authorized" Genre options than are available to pick in the Feed Manager.  I guess this is a theoretical "bug" since it ought to list everything that is actually available as a valid entry to choose from.  (???)

[https://developer.roku.com/docs/specs/search/search-feed.md](url)

A list of one or more of the following genres associated with the content item:

    action
    action sports
    adventure
    aerobics
    agriculture
    animals
    animated
    anime
    anthology
    archery
    arm wrestling
    art
    arts/crafts
    artistic gymnastics
    artistic swimming
    athletics
    auction
    auto
    auto racing
    aviation
    awards
    badminton
    ballet
    baseball
    basketball
    3x3 basketball
    beach soccer
    beach volleyball
    biathlon
    bicycle
    bicycle racing
    billiards
    biography
    blackjack
    bmx racing
    boat
    boat racing
    bobsled
    bodybuilding
    bowling
    boxing
    bullfighting
    bus./financial
    canoe
    card games
    ceremony
    cheerleading
    children
    children-music
    children-special
    children-talk
    collectibles
    comedy
    comedy drama
    community
    computers
    canoe/kayak
    consumer
    cooking
    cricket
    crime
    crime drama
    curling
    cycling
    dance
    dark comedy
    darts
    debate
    diving
    docudrama
    documentary
    dog racing
    dog show
    dog sled
    drag racing
    drama
    educational
    entertainment
    environment
    equestrian
    erotic
    event
    exercise
    fantasy
    faith
    fashion
    fencing
    field hockey
    figure skating
    fishing
    football
    food
    fundraiser
    gaelic football
    game show
    gaming
    gay/lesbian
    golf
    gymnastics
    handball
    health
    historical drama
    history
    hockey
    holiday
    holiday music
    holiday music special
    holiday special
    holiday-children
    holiday-children special
    home improvement
    horror
    horse
    house/garden
    how-to
    hunting
    hurling
    hydroplane racing
    indoor soccer
    interview
    intl soccer
    judo
    karate
    kayaking
    lacrosse
    law
    live
    luge
    martial arts
    medical
    military
    miniseries
    mixed martial arts
    modern pentathlon
    motorcycle
    motorcycle racing
    motorsports
    mountain biking
    music
    music special
    music talk
    musical
    musical comedy
    mystery
    nature
    news
    newsmagazine
    olympics
    opera
    outdoors
    parade
    paranormal
    parenting
    performing arts
    playoff sports
    poker
    politics
    polo
    pool
    pro wrestling
    public affairs
    racquet
    reality
    religious
    ringuette
    road cycling
    rodeo
    roller derby
    romance
    romantic comedy
    rowing
    rugby
    running
    rhythmic gymnastics
    sailing
    science
    science fiction
    self improvement
    shooting
    shopping
    sitcom
    skateboarding
    skating
    skeleton
    skiing
    snooker
    snowboarding
    snowmobile
    soap
    soap special
    soap talk
    soccer
    softball
    special
    speed skating
    sport climbing
    sports
    sports talk
    squash
    standup
    sumo wrestling
    surfing
    suspense
    swimming
    table tennis
    taekwondo
    talk
    technology
    tennis
    theater
    thriller
    track/field
    track cycling
    travel
    trampoline
    triathlon
    variety
    volleyball
    war
    water polo
    water skiing
    watersports
    weather
    weightlifting
    western
    wrestling
    yacht racing

**Comment by jsonnentag** [comment]
Oh, before I continue going through everything, is the Search Feed generator actually going to require and thus give and fatal error for every episode that doesn't have a language associated with it?  It would be nice to know what is considered a critical error and will just stop things from working.

Here's the Search Feed 2.0 specification that says language for every single thing isn't required and is optional.

[https://developer.roku.com/docs/specs/search/search-feed.md](url)

I circled the big IF.  I want a feed that uses the DEFAULT language and country availability and skips having that for every single episode/video.

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/f1323dd2-5a3e-4325-b6fe-947965378e3e)

**Comment by jsonnentag** [comment]
Also, do you know how the Rating setup works?  The Feed Manager seems to only allow selection of 1, but I am pretty sure I read a while ago from Roku that in order for something to be visible in other countries that country rating had to be listed, which would require multiple ratings to be available.  (???)

I think this might be a reference to it: [https://developer.roku.com/docs/specs/search/search-feed.md#ratings](url)

Ratings
**For each country supported by Roku Search**, the rating authorities, ratings, and advisory ratings are as follows:

**Comment by rrirower** [comment]
1. I will look at adding all genres to the Channel feed manager.
2. The Channel feed manager was written and available well before the demise of Direct Publisher.  One of its requirements is that feed files must follow the Roku DP specifications.  That specification says that language is required for each Episode.  I'll need to investigate using a default language in some capacity.  However, this my not be appropriate for all users.
3. As for ratings, the DP specification defined Ratings as an object and not an array.  I interpret the Seach doc to say that for each country that the Search process supports, a rating must be specified in the **Search feed**.  It does not say anything about the content feed.  Again, I would need to look into how this could be fit into the Channel feed manager.

**Comment by rrirower** [comment]
Genres resolved in version 1.0.1.3.
